pageserver/src/tenant/par_fsync.rs

```diff
@@ -53,3 +49,29 @@ pub fn par_fsync(paths: &[PathBuf]) -> io::Result<()> {
     })
     .unwrap()
 }
+
+/// Parallel fsync all files. Can be used in non-async context as it is using rayon thread pool.
+pub fn par_fsync(paths: &[PathBuf]) -> io::Result<()> {
+    const PARALLEL_PATH_THRESHOLD: usize = 1;
```

Comment:
Hm, I know this is pre-existing code, but I think it makes sense to simplify it given this constant (remove the loop, replace the check with `if paths.len() == 1 {...}`

Reply:
Note I doubt we ever use this with `paths.len() == 1`, but with two, layer file and it's parent directory, hence my comment: https://github.com/neondatabase/neon/pull/4265#pullrequestreview-1436645734

Reply:
I do believe this is a problem and I'm going to split fsync into two. First fsync all files, and then the parent directory.